module-services/service-time/AlarmOperations.cpp

```diff
@@ -345,9 +345,12 @@ namespace alarms
     auto AlarmOperationsCommon::processNextEventsQueue(const TimePoint now) -> void
     {
         if (nextSingleEvents.front()->startDate <= now) {
-            ongoingSingleEvents.insert(ongoingSingleEvents.end(),
-                                       std::make_move_iterator(nextSingleEvents.begin()),
-                                       std::make_move_iterator(nextSingleEvents.end()));
+            // if we have a critical battery level, all alarms are not triggered
+            if (!isCriticalBatteryLevelState()) {
+                ongoingSingleEvents.insert(ongoingSingleEvents.end(),
+                                           std::make_move_iterator(nextSingleEvents.begin()),
+                                           std::make_move_iterator(nextSingleEvents.end()));
+            }
```

Comment:
How about moving this block of code to a simple, well-described function? This way we would avoid adding comment:)

Reply:
done